### ISSUE TITLE
Correctly handle packages with version numbers in the library name

### ISSUE THREAD

**Issue by ennis** [mcgoo/vcpkg-rs#3]
Currently, it seems that vcpkg searches for libraries using either the package name or libraries specified with `Config::lib_name`. However, this breaks with some packages that produce a library with a version number and configuration in the filename (e.g. with the zeromq package: `libzmq-mt-4_2_3.lib`).
Would it be possible to query vcpkg for the actual library name instead of inferring it from the package name? The actual library name is present in the CMake files (`vcpkg/packages/<package-config>/share/<package-name>/`)
For now, one possible workaround for those packages is to hard-code the library name with the version and pass it to `Config::lib_name`, but this breaks as soon as the package is updated.

**Comment by mcgoo** [comment]
Thanks for the report. I hit this issue myself and I have it working for me on an branch. I'm using the `rust-zmq` crate. Would you mind trying it out and letting me know if it works for you? The Cargo.toml entry to use my copy of the rust-zmq crate would be:

```
zmq = { default-features = false, git = "https://github.com/mcgoo/rust-zmq", branch = "vcpkg"  }
```

If it works, I can get a new version of the crate pushed.

**Comment by mcgoo** [comment]
The actual vcpkg changes are at:

```
vcpkg = { git = "https://github.com/mcgoo/vcpkg-rs", branch = "deps" }
```

if you are working with some other crate.

**Comment by ennis** [comment]
Great! I noticed your fork of rust-zmq before but I tried the `vcpkg-next` branch instead of `vcpkg`. 
The `vcpkg` branch seems to work for me.

**Comment by ras0219-msft** [comment]
If possible, try to avoid depending on `packages/` -- we'd like to compress those in the future so they'll potentially move around.

The same files will be available at `$vcpkg/installed/<triplet>/share/<package-name>/` if you'd like to look at the cmake files.

Alternatively, it might be interesting to look at `$vcpkg/installed/vcpkg/info/<package-name>_*_<triplet>.list` which contains a list of all the files owned by a package. You could potentially regex-filter that list to get the libs for release and debug mode. Note that this won't follow dependencies, so it could break if we add metapackages for things like `openssl`.

**Comment by mcgoo** [comment]
Thanks @ras0219-msft. I have a version working that follows dependencies which uses the .list file. It also uses `$vcpkg/installed/vcpkg/status` which is necessary to determine dependencies (particularly in the static linking case.) Any thoughts on the wisdom of that?

**Comment by ras0219-msft** [comment]
That seems reasonable as a first cut. For an additional 0.01% correctness, You should also "concatenate" any files inside `$vcpkg/installed/vcpkg/updates` _except_ `incomplete` onto the end of status, because those are pending successful transactions.

However, I think it would perhaps make more sense if we simply offered a pkgconfig-esque mode where you could simply ask `vcpkg info <packagespec>` and we can spit out whatever you need. This could also handle feature packages, which might get somewhat difficult otherwise.

Just throwing the possibility out there, something like:
```json
{
"name": "assimp",
"triplet": "x64-windows",
"version": "xyz",
"features": ["magic", "pixie-dust"],
"depends": ["zlib", "magic"],
"files": [".... full list of full paths to directly owned files ..."]
}
```
or xml if that's easier.

**Comment by mcgoo** [comment]
Oh - that pending thing probably explains an issue I have seen where my installation has no status file on the CI server sometimes? Is there any (possibly hacky) way to force the updates to get folded in to `status` with the current version?

I actually really like the existing system because it is relatively easy to parse - vcpkg-rs ends up in a huge number of projects and I'd rather not force them to pull in `serde-json`. That said, if it is a toss-up between json and xml, then it's json for me.  

It'd be nice if it folded it all back into `status` at the end of a vcpkg run and then maybe dumped the whole status file as status.json (the info output above for all packages). I can see that might get large if it had all of the file paths in there (which is a nice thing to have) so maybe a `vcpkg info -a` to allow generating a full list on demand would be a better idea?

**Comment by mcgoo** [comment]
It looks like `vcpkg list` is enough to cause the updates to get folded in. That will be why the 'missing status' issue went away when I started to look into it!

**Comment by ras0219-msft** [comment]
Hey, it's not ~~quantum phys~~ computers without observation changing the behavior :)

It would also be easy to print the data like the current format (or heck, csv). I was thinking something extremely specialized for your purposes to start with (`vcpkg info --x-vcpkg-rs`), so it can be whatever format is most convenient. We could enable fetching the list of files with a separate command that would be newline-delimited (which we should probably have anyway).

**Comment by ras0219-msft** [comment]
Another benefit of using the existing format is that the parser translates perfectly over to handling `apt-get` on Linux if desired:
```
$ dpkg -s gcc
Package: gcc
Status: install ok installed
Priority: optional
Section: devel
Installed-Size: 44
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Source: gcc-defaults (1.150ubuntu1)
Version: 4:5.3.1-1ubuntu1
Provides: c-compiler
Depends: cpp (>= 4:5.3.1-1ubuntu1), gcc-5 (>= 5.3.1-3~)
Recommends: libc6-dev | libc-dev
Suggests: gcc-multilib, make, manpages-dev, autoconf, automake, libtool, flex, bison, gdb, gcc-doc
Conflicts: gcc-doc (<< 1:2.95.3)
Description: GNU C compiler
 This is the GNU C compiler, a fairly portable optimizing compiler for C.
 .
 This is a dependency package providing the default GNU C compiler.
Original-Maintainer: Debian GCC Maintainers <debian-gcc@lists.debian.org>
$ dpkg -L gcc
/.
/usr
/usr/bin
/usr/bin/c89-gcc
/usr/bin/c99-gcc
/usr/share
/usr/share/doc
/usr/share/doc/cpp
/usr/share/man
/usr/share/man/man1
/usr/share/man/man1/c99-gcc.1.gz
...
```

**Comment by mcgoo** [comment]
This is fixed as of 0.2.4.
